lib/matplotlib/artist.py

```diff
@@ -1494,17 +1515,30 @@ def get_setters(self):
                 continue
             func = getattr(self.o, name)
             if (not callable(func)
-                    or len(inspect.signature(func).parameters) < 2
-                    or self.is_alias(func)):
+                    or ArtistInspector.number_of_parameters(func) < 2
+                    or ArtistInspector.is_alias(func)):
                 continue
             setters.append(name[4:])
         return setters
 
-    def is_alias(self, o):
+    @staticmethod
+    @lru_cache(maxsize=None)
+    def number_of_parameters(func):
+        """Return number of parameters of function."""
+        return len(inspect.signature(func).parameters)
+
+    @staticmethod
+    @lru_cache(maxsize=None)
+    def is_alias(class_):
```

Comment:
Why is it `class_` now instead of `o`?

Reply:
I think this as a false suggestion by me. 🐑 

In practice, we only alias methods, so we should name it like that:

```suggestion
    def is_alias(method):
```